src/adapters/mc/OsConfigResource.c

```diff
@@ -1429,6 +1426,13 @@ void MI_CALL OsConfigResource_Invoke_SetTargetResource(
         FREE_MEMORY(g_procedureObjectName);
     }
 
+    if (0 != BaselineIsValidResourceIdRuleId(g_resourceId, g_ruleId, g_payloadKey, GetLog()))
+    {
+        OsConfigLogInfo(GetLog(), "[OsConfigResource.Get] BaselineIsValidResourceIdRuleId(%s, %s, %s) failed", g_resourceId, g_ruleId, g_payloadKey);
+        miResult = MI_RESULT_NOT_SUPPORTED;
+        goto Exit;
+    }
```

Comment:
Move this to Compliance lib, same as is done for ASB in the asb lib.

This is the Universal NRP adapter, it should not be changed for each baseline unless absolutely necessary.